src/modules/java.rs

```diff
@@ -55,7 +56,10 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
 fn get_java_version(context: &Context) -> Option<String> {
     let java_command = match context.get_env("JAVA_HOME") {
```

Comment:
Can you remove duplicated `java` fallback by using `map` instead?

Reply:
I've had a quick try but not seeing a clean way to do this, do you have a suggestion?

Reply:
How about this?
```rust
    let java_command = context
        .get_env("JAVA_HOME")
        .and_then(|java_home| {
            Path::new(&java_home)
                .join("bin")
                .join("java")
                .to_str()
                .map(str::to_owned)
        })
        .unwrap_or_else(|| "java".to_owned());
```

Reply:
👍🏻 much nicer than what I had

Reply:
> I've had a quick try but not seeing a nice way to do this, do you have a suggestion?

Sorry, I missed your question. I was deep in the zone trying to figure it out :upside_down_face: